src/webgl/shaders/lighting.glsl

```diff
@@ -114,7 +114,7 @@ vec3 calculateImageDiffuse( vec3 vNormal, vec3 vViewPosition ){
   vec3 worldCameraPosition =  vec3(0.0, 0.0, 0.0);  // hardcoded world camera position
   vec3 worldNormal = normalize(vNormal);
   vec2 newTexCoor = mapTextureToNormal( worldNormal );
-  vec4 texture = TEXTURE( environmentMapDiffused, newTexCoor );
+  vec4 texture = TEXTURE_CUBE( environmentMapDiffusedCubemap, newTexCoor );
```

Comment:
OH you know what the issue might be? `TEXTURE_CUBE` takes in a `vec3` as the coordinate instead of a `vec2`. The compiler probably doesn't know what to do with the combination of inputs we're giving it here, and that could be causing some more errors further on.

I think we can pass the normal directly in, like: `TEXTURE_CUBE(environmentMapDiffusedCubemap, worldNormal)`

Reply:
> OH you know what the issue might be? `TEXTURE_CUBE` takes in a `vec3` as the coordinate instead of a `vec2`. The compiler probably doesn't know what to do with the combination of inputs we're giving it here, and that could be causing some more errors further on.
> 
> I think we can pass the normal directly in, like: `TEXTURE_CUBE(environmentMapDiffusedCubemap, worldNormal)`

Yes, you're correct! In many GLSL implementations, the textureCube function takes a vec3 as the first argument, representing the 3D direction vector in the cube map space. So, using `TEXTURE_CUBE(environmentMapDiffusedCubemap, worldNormal)` is likely the correct way to sample from the cube map based on the normal vector.
Thanks a lot for looking into this @davepagurek !!